src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let clone = Object.assign({}, state);
+  for (let i = 0; i < actions.length; i++) {
+    const { type, extraData, keysToRemove } = actions[i];
+
+    if (i > 0) {
+      clone = Object.assign({}, newState[i - 1]);
+    }
```

Comment:
This reassignment of 'clone' is unnecessary. The initial clone of the state is correct, but you should modify the 'clone' directly in each action and then push it to 'newState' after the action is processed. This will avoid additional unnecessary cloning for each action.

Reply:
I believe it's a necessary action because the task itself called transformStateWithClones so I populate the array with clones regardless of actions, otherwise it will be duplicated code.